onnxscript/ir/passes/common/lift_constants_to_initializers.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Lift constants to initializers."""
+
+from __future__ import annotations
+
+__all__ = [
+    "LiftConstantsToInitializersPass",
+]
+
+import logging
+
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+
+
+class LiftConstantsToInitializersPass(ir.passes.InPlacePass):
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
+        """Convert constant nodes in main graph to initializers."""
+        count = 0
+        for node in model.graph:
```

Comment:
This won't process nodes within subgraphs (in an If node, for example). This also introduces a design question: should we move the constant nodes from an If node subgraph into the main graph's initializer? The tricky part is that it could introduce name conflicts ... if we have different tensors in the then-branch and else-branch which have same name. I think the IR naming may handle it correctly, but worth checking.

Reply:
I think we should avoid subgraphs, for now. The constant nodes in subgraphs tend to not be model weights

Reply:
The second part is right ... but why do you say we should avoid subgraphs for now? Eg., one option would be to simply replace the for loop to iterate over all nodes (I believe there is an iterator for that, right?). Do you mean the naming part could be challenging?

Reply:
Actually I suppose it’s ok doing it either way. Initially I thought the naming part can be challenging, but since we already have some knowledge of the used names in the main graph, pulling values from subgraphs and giving them new names should be safe

Reply:
In that case you may want to make the subgraph constants to subgraph initializers first, and use the pass #2157 to lift them from subgraphs to the main graph. 

To find the owning graph of the node you can do node.graph.

Reply:
The PR supports lifting subgraph constants to initializers as well now. We will have another pass to lift subgraph initializers to main graph if it's necessary.